### PR TITLE
8287200: Test java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java timed out after JDK-8287103

### PR DESCRIPTION
**Pull request by lmesnik**
Need to use proper synchronization.

The CyclicBarriers might move the thread to WAITING state but not BLOCKED. So it should not confuse existing checks.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.java.net/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287200](https://bugs.openjdk.java.net/browse/JDK-8287200): Test java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java timed out after  JDK-8287103

### Reviewers
 * [Kevin Walls](https://openjdk.java.net/census#kevinw) (@kevinjwalls - Committer) ⚠️ Review applies to [3402eb82](https://git.openjdk.java.net/jdk/pull/8874/files/3402eb82380ce7e92581eace2b35e6e03a8e48e5)
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**) ⚠️ Review applies to [3402eb82](https://git.openjdk.java.net/jdk/pull/8874/files/3402eb82380ce7e92581eace2b35e6e03a8e48e5)
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8874/head:pull/8874` \
`$ git checkout pull/8874`

Update a local copy of the PR: \
`$ git checkout pull/8874` \
`$ git pull https://git.openjdk.java.net/jdk pull/8874/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8874`

View PR using the GUI difftool: \
`$ git pr show -t 8874`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8874.diff">https://git.openjdk.java.net/jdk/pull/8874.diff</a>

</details>

### CODE REVIEW

**Review by kevinjwalls — APPROVED** [review]
Yes I think that's nice - the two competing Threads own their first lock and wait until the other thread has owned its first lock, before continuing. 8-)

**Review by plummercj — APPROVED** [review]

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java [hunk 1]
```diff
@@ -98,6 +97,9 @@ public static void main(String[] args) throws Exception {
     private static void awaitBlocked(Thread thread) throws InterruptedException {
         while (thread.getState() != Thread.State.BLOCKED) {
             Thread.sleep(10);
+            if (thread.getState() == Thread.State.TERMINATED) {
```
Comment (master-code-java) [inline test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:100]: Thread.State.TERMINATED == thread.getState(). Does it make sense?

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java [hunk 2]
```diff
@@ -98,6 +97,9 @@ public static void main(String[] args) throws Exception {
     private static void awaitBlocked(Thread thread) throws InterruptedException {
         while (thread.getState() != Thread.State.BLOCKED) {
```
Comment (master-code-java) [inline test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:98]: Thread.State.BLOCKED == thread.getState(). Does it make sense?

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java (hunk 1 above)
Comment (lmesnik) [reply test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:100]: Not sure. What is the goal?

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java (hunk 2 above)
Comment (lmesnik) [reply test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:98]: Do you mean '==' or '!='? Both don't make sense to me, honestly. Could you please elaborate?

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java (hunk 1 above)
Comment (AlanBateman) [reply test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:100]: I assume this test is catch the catch where the thread terminates without blocking. Using isAlive might be cleaner for people to understand.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**